src/public-modules/Leaderboard/sagas.js

```diff
@@ -1,17 +1,60 @@
 import request from 'utils/request';
-import { call, put, takeLatest } from 'redux-saga/effects';
+import { all, call, put, takeLatest, select } from 'redux-saga/effects';
 import { actionTypes, actions } from 'public-modules/Leaderboard';
-import { LIMIT, PLATFORM } from './constants';
+import { leaderboardQuerySelector, leaderboardSelector } from './selectors';
+import { rootLeaderboardUISelector } from 'containers/Leaderboard/selectors';
 
-const { LOAD_LEADERBOARD } = actionTypes;
-const { loadLeaderboardFail, loadLeaderboardSuccess } = actions;
+const { LOAD_LEADERBOARD, LOAD_MORE_LEADERBOARD } = actionTypes;
+const {
+  loadLeaderboardFail,
+  loadLeaderboardSuccess,
+  loadMoreLeaderboardSuccess
+} = actions;
+
+export function* loadLeaderboard() {
+  const params = yield select(leaderboardQuerySelector);
 
-export function* loadLeaderboard(action) {
-  let { leaderboardCategory } = action;
   try {
-    let endpoint = `leaderboard/${leaderboardCategory}?limit=${LIMIT}&platform__in=${PLATFORM}`;
-    const leaderboard = yield call(request, endpoint, 'GET');
-    yield put(loadLeaderboardSuccess(leaderboard));
+    const { issuer, fulfiller } = yield all({
+      issuer: call(request, `leaderboard/issuer/`, 'GET', { params }),
+      fulfiller: call(request, `leaderboard/fulfiller/`, 'GET', { params })
+    });
+
+    yield put(loadLeaderboardSuccess({ issuer, fulfiller }));
+  } catch (e) {
+    yield put(loadLeaderboardFail(e));
+  }
+}
+
+export function* loadMoreLeaderboard() {
+  const params = yield select(leaderboardQuerySelector);
+  const { leaderboard } = yield select(leaderboardSelector);
+  const { toggleValue } = yield select(rootLeaderboardUISelector);
+  const offset = {
+    issuer: leaderboard.issuer.length,
+    fulfiller: leaderboard.fulfiller.length
+  };
+
+  const getData = (toggleValue, offset, type) => {
+    return toggleValue == type
+      ? call(request, `leaderboard/${type}/`, 'GET', {
+          params: { ...params, offset: offset[type] }
+        })
+      : [];
+  };
+
+  const { issuer, fulfiller } = yield all({
+    issuer: getData(toggleValue, offset, 'issuer'),
+    fulfiller: getData(toggleValue, offset, 'fulfiller')
+  });
+
+  try {
+    yield put(
+      loadMoreLeaderboardSuccess({
+        issuer: issuer.results || [],
+        fulfiller: fulfiller.results || []
+      })
+    );
   } catch (e) {
     yield put(loadLeaderboardFail(e));
```

Comment:
yeah - so this error is an ACTION error - not a page error. I have not created the pattern yet - so we'll need to handle this error differently when the pattern is there. We can talk about this soon.

Reply:
👍